components/sup/src/util/pkg.rs

```diff
@@ -125,11 +126,46 @@ pub async fn install_channel_head(url: &str,
     install_no_ui(url, &channel_latest_ident.into(), channel).await
 }
 
+/// Uninstall all but the `number_latest_to_keep` packages.
+///
+/// Returns the number of packages that were uninstalled
+pub async fn uninstall_all_but_latest(ident: impl AsRef<PackageIdent>,
+                                      number_latest_to_keep: usize)
+                                      -> HabResult<usize> {
+    let ident = ident.as_ref().clone();
+    let mut idents = list::package_list(&ident.into())?;
+    if number_latest_to_keep >= idents.len() {
+        return Ok(0);
+    }
+    idents.reverse(); // The packages to keep now occur first in the list
```

Comment:
It would be safer to explicitly order these how we want them, rather than implicitly rely on them being given to us in a particular order already.

I'd hate for `list::package_list` to change at some point in the future and then we suddenly start rolling people back to the earliest service on their machine and prevent them from ever updating it :grimacing: 

(Alternatively, make it painfully clear in the documentation for `list::package_list` that the contract for that function includes the order the results are returned in, along with a test that validates that behavior.)

Reply:
Explicitly sorting is much safer, thanks. Addressed in [a0ce28](https://github.com/habitat-sh/habitat/pull/7582/commits/a0ce289bcaa98503aaba027293e72427e64ef570)